pkg/provision/engine.go

```diff
@@ -867,10 +868,44 @@ func (e *NativeEngine) uninstallDeployment(ctx context.Context, dl *gridtypes.De
 
 }
 
+func getZmountSize(wl *gridtypes.WorkloadWithID) (gridtypes.Unit, error) {
+	dataI, err := wl.WorkloadData()
+	if err != nil {
+		return 0, err
+	}
+	zmount, ok := dataI.(*zos.ZMount)
+	if !ok {
+		return 0, fmt.Errorf("failed to get workload data as a zmount '%v'", dataI)
+	}
+	return zmount.Size, nil
+}
+
+func sortZmountWorkloads(workloads []*gridtypes.WorkloadWithID) {
+	sort.Slice(workloads, func(i, j int) bool {
+		sizeI, err := getZmountSize(workloads[i])
+		if err != nil {
+			log.Error().Err(err).Stringer("id", workloads[i].ID).Msg("failed to get zmount size")
+			return false
+		}
+
+		sizeJ, err := getZmountSize(workloads[j])
+		if err != nil {
+			log.Error().Err(err).Stringer("id", workloads[j].ID).Msg("failed to get zmount size")
+			return true
```

Comment:
Why do u return true here but false in the first case ? just wondering. I think you should return false always in case if both elements are `equal`.